### ISSUE TITLE
[Bug] Golang CI failed

### ISSUE THREAD

**Issue by zuston** [apache/incubator-uniffle#512]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

```
# golang.org/x/sys/unix
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/syscall.go:83:16: undefined: unsafe.Slice
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/syscall_linux.go:2270:9: undefined: unsafe.Slice
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/syscall_unix.go:118:7: undefined: unsafe.Slice
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/sysvshm_unix.go:33:7: undefined: unsafe.Slice

```
Detailed info could be found in https://github.com/apache/incubator-uniffle/actions/runs/4031027397/jobs/6930075459

### Affects Version(s)

master

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by zuston** [comment]
@advancedxy @wangao1236 could you help check this?

**Comment by kaijchen** [comment]
It says "note: module requires Go 1.17" here:
https://github.com/apache/incubator-uniffle/actions/runs/4031027397/jobs/6930075459#step:6:3421

It's probably caused by transitive dependency of this package:
https://pkg.go.dev/golang.org/x/sys@v0.4.0/unix

**Comment by kaijchen** [comment]
The easiest fix is to bump golang to 1.17 and update the go.mod.
@jerqi Is it OK to bump the golang version?
